Automattic/jetpack
# Add Custom Autoloader

This PR creates a custom autoloader package that could be used by a number of different plugins
to always load the latest version of the package libraries. 

The Readme has more info. 

This PR is a mash up of https://github.com/Automattic/jetpack/pull/12295 and https://github.com/schlessera/composer-wp-autoload/

#### Changes proposed in this Pull Request:
In this PR we create a new composer plugin that helps generate a few new files in the vender directory. 

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here.

#### Testing instructions:
* `$ yarn docker:sh`
* `$ cd /var/www/html/wp-content/plugins/jetpack/`
* `$ composer install`

Does the site still load as expected.

Test it actually loads the more up to date version of the package
* first good luck
* `$ yarn docker:sh`
* Update Logo package version to 1.0.0
* * add `"version": "1.0.0"` to `packages/logo/composer.json` inside the Jetpack plugin folder
* Modify `packages/logo/src/Logo.php`
* * Make it render something that is distinctive for this version
* run `composer install` for `wp-content/plugins/a/`
* Update Logo package version to 1.0.1
* * add `"version": "1.0.1"` to `packages/logo/composer.json` inside the Jetpack plugin folder
* Modify `packages/logo/src/Logo.php`
* * Make it render something that is distinctive for this version
* run `composer install` for `wp-content/plugins/b/`
* activate both `a` and `b`
* check that your site posts render the distinct 1.0.1 version of the package




#### Proposed changelog entry for your changes:

*
